pkg/apis/gateway/v1alpha1/types.go

```diff
@@ -61,14 +59,24 @@ type GatewaySpec struct {
 	// Version is used for marking event version
 	Version string `json:"version" protobuf:"bytes,6,opt,name=version"`
 
-	// Service is the name of the service to expose the gateway
-	Service Service `json:"service,omitempty" protobuf:"bytes,7,opt,name=service"`
+	// ServiceSpec is the specifications of the service to expose the gateway
+	ServiceSpec *corev1.ServiceSpec `json:"serviceSpec,omitempty" protobuf:"bytes,7,opt,name=serviceSpec"`
 
 	// Sensors are list of sensors to dispatch events to
 	Sensors []string `json:"sensors" protobuf:"bytes,8,opt,name=sensors"`
 
 	// ServiceAccountName is name of service account to run the gateway
-	ServiceAccountName string `json:"serviceAccountName" protobuf:"bytes,9,opt,name=serviceAccountName"`
+	ServiceAccountName string `json:"serviceAccountName,omitempty" protobuf:"bytes,9,opt,name=serviceAccountName"`
+
+	// Affinity for scheduling
+	Affinity *corev1.Affinity `json:"affinity,omitempty" protobuf:"bytes,10,opt,name=affinity"`
+
+	// ImagePullPolicy for pulling the image
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty" protobuf:"bytes,11,opt,name=imagePullPolicy,casttype=k8s.io/api/core/v1.PullPolicy"`
+
+	// todo: do we need separate labels or just use same labels defined on gateway resource?
```

Comment:
i don't think we need separate labels. we get labels from the serviecSpec and from the gateway metadata. just need to be clear in the specification/declaration where each one applies.